android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,58 +35,64 @@ class MediaSessionFullscreenFeature(
-    private fun processFullscreen(sessionStates: List<SessionState>) {
-        /* there should only be one fullscreen session */
-        val activeState = sessionStates.firstOrNull()
-        if (activeState == null || activeState.mediaSessionState?.fullscreen != true) {
-            activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
-            return
-        }
+    private fun setOrientationForTabState(activeTabState: SessionState) {
+        when (activeTabState.mediaSessionState?.elementMetadata?.portrait) {
+            true ->
+                activity.requestedOrientation =
+                    ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
 
-        if (store.state.findCustomTabOrSelectedTab(tabId)?.id == activeState.id) {
-            when (activeState.mediaSessionState?.elementMetadata?.portrait) {
-                true ->
+            false ->
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInPictureInPictureMode) {
+                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+                } else {
                     activity.requestedOrientation =
-                        ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
-                false ->
-                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInPictureInPictureMode) {
-                        activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
-                    } else {
-                        activity.requestedOrientation =
-                            ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
-                    }
-                else -> activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
-            }
+                        ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                }
+
+            null -> activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
         }
     }
 
-    private fun processDeviceSleepMode(sessionStates: List<SessionState>) {
-        val activeTabState = sessionStates.firstOrNull()
-        if (activeTabState == null || activeTabState.mediaSessionState?.fullscreen != true) {
-            activity.window?.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
-            return
-        }
+    private fun setDeviceSleepModeForTabState(activeTabState: SessionState) {
         activeTabState.mediaSessionState?.let {
             when (activeTabState.mediaSessionState?.playbackState) {
                 MediaSession.PlaybackState.PLAYING -> {
                     activity.window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
                 }
+
                 else -> {
                     activity.window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
                 }
             }
         }
     }
 
+    private fun SessionState?.hasSameOrientationInformationAs(other: SessionState?): Boolean =
```

Comment:
Love this! I had been playing around with this function signature a bit myself and didn't find anything nearly this good ☺️